src/styles/calendar.scss

```diff
@@ -0,0 +1,36 @@
+.calendar {
+  position: relative;
+  width: 800px;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  margin: 0 auto;
+  padding-left: 47px;
+
+  &__day {
+    background-color: #eee;
+    width: $cell-size;
+    height: $cell-size;
+    border: 1px solid black;
+    position: relative;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    transition: transform 0.5s;
+    cursor: pointer;
+
+    @for $i from 1 through 31 {
+      &[data-day='#{$i}']::before {
+        content: '#{$i}';
+        font-size: 30px;
+        font-family: Arial, Helvetica, sans-serif;
+      }
```

Comment:
The loop for generating days should not be placed inside the `.calendar__day` block as it may lead to code duplication. Instead, consider moving the loop outside of this block to dynamically generate `.calendar__day` elements. This will also help avoid the unnecessary specificity of the `[data-day]` attribute selector.